--- rpmbuild/SPECS/nginx.spec	2016-05-31 14:08:45.000000000 +0000
+++ nginx-libressl.spec	2016-07-08 16:18:33.753809017 +0000
@@ -1,3 +1,4 @@
+%define libressl_version 2.3.6
 #
 %define nginx_home %{_localstatedir}/cache/nginx
 %define nginx_user nginx
@@ -120,6 +121,8 @@
         --with-mail_ssl_module \
         --with-file-aio \
         --with-ipv6 \
+        --with-openssl="%{_sourcedir}/portable-%{libressl_version}" \
+        --with-ld-opt="-lrt" \
         %{?with_http2:--with-http_v2_module}")

 Summary: High performance web server
@@ -209,6 +212,12 @@
 Dynamic nJScript module for nginx.

 %prep
+cd %{_sourcedir}
+wget "https://github.com/libressl-portable/portable/archive/v%{libressl_version}.tar.gz" -O %{_sourcedir}/v%{libressl_version}.tar.gz
+tar -xzf v%{libressl_version}.tar.gz
+cd portable-%{libressl_version}
+./autogen.sh
+
 %setup -q
 tar xvzf %SOURCE13
 cp %{SOURCE2} .
